### ISSUE TITLE
Question abount the implementation of "gradient_accumulate_every" 

### ISSUE THREAD

**Issue by CyberPegasus** [lucidrains/denoising-diffusion-pytorch#250]
I'm wondering that, referring to the 1016th line in denoising_diffusion_pytorch.py, i.e., "self.accelerator.backward(loss)", should it be placed outside the loop "for _ in range(self.gradient_accumulate_every): ....." ?
In other words, does "self.gradient_accumulate_every" denote the hyperparameter to spilit a batch ?

**Comment by CyberPegasus** [comment]
And, in this way, "self.accelerator.backward(total_loss)" will be conducted.

**Comment by CyberPegasus** [comment]
I understand the accumulation of gradient in several mini-batches is also a way to increase the equivalent batch size.
